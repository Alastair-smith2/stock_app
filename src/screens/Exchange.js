import React, { Component } from "react";
import { View, Text, StyleSheet, FlatList, Alert } from "react-native";
import Input from "../components/Input";
import Search from "../components/Search";
import Instructions from "../components/Instructions";

export class Exchange extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currencyOne: "",
      currencyTwo: ""
    };
    this.handleTextChange = this.handleTextChange.bind(this);
    this.handleSearch = this.handleSearch.bind(this);
  }

  handleTextChange = (type, text) => {
    this.setState({ [type]: text });
  };

  handleSearch = () => {
    if (!this.state.currencyOne == "" && !this.state.currencyTwo == "") {
      Alert.alert("Fetch to come");
    } else {
      Alert.alert("Please make sure you've enntered a value for each currency");
    }
  };

  render() {
    return (
      <View style={styles.container}>
        <Instructions
          instructionContainer={styles.instructionContainer}
          label={"Please enter 3 letter currency symbols"}
        />
        <View style={styles.searchContainer}>
          <Search
            error={false}
            searchType={"currency"}
            currencyOne={this.state.currencyOne}
            typeOne="currencyOne"
            labelOne={"From"}
            currencyTwo={this.state.currencyTwo}
            typeTwo="currencyTwo"
            labelTwo={"To"}
            submitSearch={this.handleSearch}
            onTextSearch={this.handleTextChange}
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff"
  },
  instructionContainer: {
    flex: 0.2,
    justifyContent: "center",
    alignItems: "center"
  },
  searchContainer: {
    flex: 0.3,
    margin: 10
  },
  center: {
    flex: 1
  }
});

Exchange.navigationOptions = {
  title: "Exchange Rate",
  headerStyle: {
    backgroundColor: "#EA572D",
    paddingBottom: 10
  },
  headerTitleStyle: {
    color: "#FFF",
    fontSize: 20,
    flex: 1,
    textAlign: "center"
  },
  headerTintColor: "#fff",
  headerLeft: null
};

export default Exchange;
