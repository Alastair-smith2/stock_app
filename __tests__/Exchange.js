import React from "react";
import { shallow } from "enzyme";
import { Alert } from "react-native";
import { Exchange } from "../src/screens/Exchange.js";

describe("The exchange tab", () => {
  let wrapper;
  let navigation = {
    navigate: jest.fn()
  };
  jest.mock("Alert", () => {
    return {
      alert: jest.fn()
    };
  });
  const spy = jest.spyOn(Alert, "alert");
  beforeEach(() => {
    wrapper = shallow(<Exchange navigation={navigation} />);
    navigation.navigate.mockReset();
  });
  describe("It should render", () => {
    it("Should render", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe("The features of the exchange tab", () => {
    it("Should have a function that handles text change", () => {
      wrapper.instance().handleTextChange("currencyOne", "GBP");
      expect(wrapper.state("currencyOne")).toEqual("GBP");
    });

    it("Should have a function that handles a submit change", () => {
      wrapper.instance().handleSearch();
      expect(Alert.alert.mock.calls.length).toEqual(1);
      wrapper.setState({ currencyOne: "GBP", currencyTwo: "USD" });
      wrapper.instance().handleSearch();
      expect(Alert.alert.mock.calls.length).toEqual(2);
      expect(Alert.alert.mock.calls[1][0]).toEqual("Fetch to come");
    });
  });
});
