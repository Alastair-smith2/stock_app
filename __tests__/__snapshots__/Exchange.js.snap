// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The exchange tab It should render Should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Exchange
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Instructions
          instructionContainer={
            Object {
              "alignItems": "center",
              "flex": 0.2,
              "justifyContent": "center",
            }
          }
          label="Please enter 3 letter currency symbols"
        />,
        <View
          style={
            Object {
              "flex": 0.3,
              "margin": 10,
            }
          }
        >
          <Search
            currencyOne=""
            currencyTwo=""
            error={false}
            labelOne="From"
            labelTwo="To"
            onTextSearch={[Function]}
            searchType="currency"
            submitSearch={[Function]}
            typeOne="currencyOne"
            typeTwo="currencyTwo"
          />
        </View>,
      ],
      "style": Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "instructionContainer": Object {
            "alignItems": "center",
            "flex": 0.2,
            "justifyContent": "center",
          },
          "label": "Please enter 3 letter currency symbols",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Search
            currencyOne=""
            currencyTwo=""
            error={false}
            labelOne="From"
            labelTwo="To"
            onTextSearch={[Function]}
            searchType="currency"
            submitSearch={[Function]}
            typeOne="currencyOne"
            typeTwo="currencyTwo"
          />,
          "style": Object {
            "flex": 0.3,
            "margin": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "currencyOne": "",
            "currencyTwo": "",
            "error": false,
            "labelOne": "From",
            "labelTwo": "To",
            "onTextSearch": [Function],
            "searchType": "currency",
            "submitSearch": [Function],
            "typeOne": "currencyOne",
            "typeTwo": "currencyTwo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Instructions
            instructionContainer={
              Object {
                "alignItems": "center",
                "flex": 0.2,
                "justifyContent": "center",
              }
            }
            label="Please enter 3 letter currency symbols"
          />,
          <View
            style={
              Object {
                "flex": 0.3,
                "margin": 10,
              }
            }
          >
            <Search
              currencyOne=""
              currencyTwo=""
              error={false}
              labelOne="From"
              labelTwo="To"
              onTextSearch={[Function]}
              searchType="currency"
              submitSearch={[Function]}
              typeOne="currencyOne"
              typeTwo="currencyTwo"
            />
          </View>,
        ],
        "style": Object {
          "backgroundColor": "#fff",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "instructionContainer": Object {
              "alignItems": "center",
              "flex": 0.2,
              "justifyContent": "center",
            },
            "label": "Please enter 3 letter currency symbols",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Search
              currencyOne=""
              currencyTwo=""
              error={false}
              labelOne="From"
              labelTwo="To"
              onTextSearch={[Function]}
              searchType="currency"
              submitSearch={[Function]}
              typeOne="currencyOne"
              typeTwo="currencyTwo"
            />,
            "style": Object {
              "flex": 0.3,
              "margin": 10,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "currencyOne": "",
              "currencyTwo": "",
              "error": false,
              "labelOne": "From",
              "labelTwo": "To",
              "onTextSearch": [Function],
              "searchType": "currency",
              "submitSearch": [Function],
              "typeOne": "currencyOne",
              "typeTwo": "currencyTwo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
